Userland/printf.cpp

```diff
@@ -0,0 +1,297 @@
+    ALWAYS_INLINE unsigned operator()(V arg) const
+    {
+        if (arg.argc == 0)
+            return 0;
+
+        auto result = *arg.argv++;
+        --arg.argc;
+        return strtoul(result, nullptr, 10);
+    }
+};
+
+template<typename V>
+struct ArgvNextArgument<i64, V> {
+    ALWAYS_INLINE i64 operator()(V arg) const
+    {
+        if (arg.argc == 0)
+            return 0;
+
+        auto result = *arg.argv++;
+        --arg.argc;
+        return strtoll(result, nullptr, 10);
+    }
+};
+
+template<typename V>
+struct ArgvNextArgument<u64, V> {
+    ALWAYS_INLINE u64 operator()(V arg) const
+    {
+        if (arg.argc == 0)
+            return 0;
+
+        auto result = *arg.argv++;
+        --arg.argc;
+        return strtoull(result, nullptr, 10);
+    }
+};
+
+template<typename V>
+struct ArgvNextArgument<double, V> {
+    ALWAYS_INLINE double operator()(V arg) const
+    {
+        if (arg.argc == 0)
+            return 0;
+
+        auto result = *arg.argv++;
+        --arg.argc;
+        return strtod(result, nullptr);
+    }
+};
+
+template<typename V>
+struct ArgvNextArgument<int*, V> {
+    ALWAYS_INLINE int* operator()(V) const
+    {
+        ASSERT_NOT_REACHED();
+        return nullptr;
+    }
+};
+
+struct ArgvWithCount {
+    char**& argv;
+    int& argc;
+};
+
+String handle_escapes(const char* string)
+{
+    StringBuilder builder;
+    for (auto c = *string; c; c = *++string) {
+        if (c == '\\') {
+            if (string[1]) {
+                switch (c = *++string) {
+                case '\\':
+                case '"':
+                    builder.append(c);
+                    break;
+                case 'a':
+                    builder.append('\a');
+                    break;
+                case 'b':
+                    builder.append('\b');
+                    break;
+                case 'c':
+                    goto end;
+                case 'e':
+                    builder.append('\e');
+                    break;
+                case 'f':
+                    builder.append('\f');
+                    break;
+                case 'n':
+                    builder.append('\n');
+                    break;
+                case 'r':
+                    builder.append('\r');
+                    break;
+                case 't':
+                    builder.append('\t');
+                    break;
+                case 'v':
+                    builder.append('\v');
+                    break;
+                case 'x':
+                    fail("Unsupported escape '\\x'");
+                case 'u':
+                    fail("Unsupported escape '\\x'");
+                case 'U':
+                    fail("Unsupported escape '\\x'");
+                default:
+                    builder.append(c);
+                }
+            } else {
+                builder.append(c);
+            }
+        } else {
+            builder.append(c);
+        }
+    }
+
+end:;
```

Comment:
:thinking:

I'd just say `return builder.build()` in that one place above.

Reply:
Alright.
I honestly didn't expect `%c` to just terminate the format, so I threw in a `goto end` on instinct lol